frontend/src/metabase/collections/components/CollectionSidesheet/CollectionSidesheet.tsx

```diff
@@ -0,0 +1,134 @@
+import { useCallback, useMemo, useState } from "react";
+import { useMount } from "react-use";
+import { t } from "ttag";
+
+import { Sidesheet, SidesheetCard } from "metabase/common/components/Sidesheet";
+import { EntityIdCard } from "metabase/components/EntityIdCard";
+import EditableText from "metabase/core/components/EditableText";
+import { color } from "metabase/lib/colors";
+import type { ObjectWithModel } from "metabase/lib/icon";
+import { PLUGIN_COLLECTIONS } from "metabase/plugins";
+import {
+  Box,
+  Group,
+  FixedSizeIcon as Icon,
+  Stack,
+  Text,
+  Title,
+} from "metabase/ui";
+import type { Collection } from "metabase-types/api";
+
+export const CollectionSidesheet = ({
+  onClose,
+  collection,
+  onUpdateCollection,
+}: {
+  onClose: () => void;
+  collection: Collection;
+  onUpdateCollection: (entity: Collection, values: Partial<Collection>) => void;
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  useMount(() => {
+    // this component is not rendered until it is "open"
+    // but we want to set isOpen after it mounts to get
+    // pretty animations
+    setIsOpen(true);
+  });
+
+  const iconProps = useMemo(() => {
+    const icon = PLUGIN_COLLECTIONS.getIcon({
+      ...collection,
+      model: "collection",
+    } as ObjectWithModel);
+    if (icon.color) {
+      icon.color = color(icon.color);
+    }
+    return icon;
+  }, [collection]);
+
+  const handleChangeDescription = useCallback(
+    (description: string) => {
+      onUpdateCollection(collection, { description: description || null });
+    },
+    [collection, onUpdateCollection],
+  );
+
+  return (
+    <Sidesheet
+      title={t`Info`}
+      isOpen={isOpen}
+      data-testid="collection-sidesheet"
+      size="md"
+      onClose={onClose}
+    >
+      <Stack spacing="lg">
+        <SidesheetCard pb="md">
+          <Stack spacing="md">
+            <Stack spacing="xs">
+              <Title lh={1} size="sm" color="text-light">
+                {t`Description`}
+              </Title>
+              <EditableDescription
+                collection={collection}
+                handleChangeDescription={handleChangeDescription}
+              />
+            </Stack>
+            {collection.authority_level === "official" && (
+              <Group noWrap spacing="sm" pb="xs">
+                <Icon {...iconProps} />
+                <Text lh={1}>{t`Official collection`}</Text>
+              </Group>
+            )}
```

Comment:
it's a pain, but this is enterprise code that needs to stay out of the open source codebase.